In [2]:
!wget https://github.com/alexeygrigorev/datasets/raw/refs/heads/master/bank-full.csv

--2024-10-23 19:27:21--  https://github.com/alexeygrigorev/datasets/raw/refs/heads/master/bank-full.csv
Resolving github.com (github.com)... 64:ff9b::14cd:f3a6, 20.205.243.166
Connecting to github.com (github.com)|64:ff9b::14cd:f3a6|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/alexeygrigorev/datasets/refs/heads/master/bank-full.csv [following]
--2024-10-23 19:27:22--  https://raw.githubusercontent.com/alexeygrigorev/datasets/refs/heads/master/bank-full.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 2606:50c0:8000::154, 2606:50c0:8001::154, 2606:50c0:8002::154, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|2606:50c0:8000::154|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4610348 (4.4M) [text/plain]
Saving to: ‘bank-full.csv’

bank-full.csv       100%[===================>]   4.40M  9.84MB/s    in 0.4s    

2024-10-23 19:27:23 (9.84 MB/s) - ‘ban

In [3]:
import pandas as pd     

In [4]:
df = pd.read_csv('bank-full.csv')
df.head()

age;"job";"marital";"education";"default";"balance";"housing";"loan";"contact";"day";"month";"duration";"campaign";"pdays";"previous";"poutcome";"y"
0  58;"management";"married";"tertiary";"no";2143...                                                                                                  
1  44;"technician";"single";"secondary";"no";29;"...                                                                                                  
2  33;"entrepreneur";"married";"secondary";"no";2...                                                                                                  
3  47;"blue-collar";"married";"unknown";"no";1506...                                                                                                  
4  33;"unknown";"single";"unknown";"no";1;"no";"n...

## Question 1
pipenv, version 2024.2.0

## Question 2
455b4659c7c6ebc9746ce2c17e0423d9b60ac22832ebbccebf6ab1c265fb8ea6

In [8]:
!wget https://raw.githubusercontent.com/DataTalksClub/machine-learning-zoomcamp/master/cohorts/2024/05-deployment/homework/model1.bin
!wget https://raw.githubusercontent.com/DataTalksClub/machine-learning-zoomcamp/master/cohorts/2024/05-deployment/homework/dv.bin

--2024-10-23 19:38:31--  https://raw.githubusercontent.com/DataTalksClub/machine-learning-zoomcamp/master/cohorts/2024/05-deployment/homework/model1.bin
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 2606:50c0:8000::154, 2606:50c0:8001::154, 2606:50c0:8002::154, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|2606:50c0:8000::154|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 850 [application/octet-stream]
Saving to: ‘model1.bin’

model1.bin          100%[===================>]     850  --.-KB/s    in 0s      

2024-10-23 19:38:31 (18.4 MB/s) - ‘model1.bin’ saved [850/850]

--2024-10-23 19:38:31--  https://raw.githubusercontent.com/DataTalksClub/machine-learning-zoomcamp/master/cohorts/2024/05-deployment/homework/dv.bin
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 2606:50c0:8001::154, 2606:50c0:8002::154, 2606:50c0:8003::154, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)

## Question 3

In [9]:
import pickle

with open('model1.bin', 'rb') as file:
    model = pickle.load(file)

with open('dv.bin', 'rb') as file:
    dv = pickle.load(file)

# Example input data
input_data = {"job": "management", "duration": 400, "poutcome": "success"}

# Transform the input data using the DictVectorizer
X = dv.transform([input_data])

# Make a prediction using the LogisticRegression model
prediction = model.predict(X)
prediction_proba = model.predict_proba(X)

print(f'Prediction: {prediction[0]}')
print(f'Prediction Probability: {prediction_proba[0]}')

Prediction: 1
Prediction Probability: [0.24090335 0.75909665]


/Users/riodpp/.pyenv/versions/3.10.0/envs/sandbox/lib/python3.10/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator LogisticRegression from version 1.5.2 when using version 1.5.1. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(
/Users/riodpp/.pyenv/versions/3.10.0/envs/sandbox/lib/python3.10/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DictVectorizer from version 1.5.2 when using version 1.5.1. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(


The probability of that client will get a subscription is 0.759

In [10]:
!md5sum model1.bin dv.bin

3d8bb28974e55edefa000fe38fd3ed12  model1.bin
7d37616e00aa80f2152b8b0511fc2dff  dv.bin


## Question 4
Output
{"prediction":0,"prediction_probability":[0.6651929652448895,0.33480703475511053]}

## Question 5
svizor/zoomcamp-model              3.11.5-slim          975e7bdca086   4 days ago      130MB

## Question 6
Output
{'prediction': 1, 'prediction_probability': [0.2409033483120342, 0.7590966516879658]}